validator/sawtooth_validator/journal/block_validator.py

```diff
@@ -140,8 +143,14 @@ def __init__(self,
         self._config_dir = config_dir
         self._permission_verifier = permission_verifier
 
-        self._validation_rule_enforcer = \
-            ValidationRuleEnforcer(SettingsViewFactory(state_view_factory))
+        self._validation_rule_enforcer = ValidationRuleEnforcer(
+            SettingsViewFactory(state_view_factory))
+
+        self._thread_pool = InstrumentedThreadPoolExecutor(1) \
+            if thread_pool is None else thread_pool
+
+    def stop(self):
+        self._thread_pool.shutdown(wait=True)
```

Comment:
This thread pool should only be shutdown by this class if it was created by this class.  Otherwise, the creator of this class may end up having a shutdown threadpool that is used in other situations.

Reply:
The threadpool is owned by the class. I just needed a way to pass in the synchronous executor for the unit tests. Unless you know another way to swap out the thread pool without making it an argument to init.